src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -440,13 +440,15 @@ protected final void retrieve(@CheckForNull SCMSourceCriteria criteria,
             String fullName = repoOwner + "/" + repository;
             final GHRepository repo = github.getRepository(fullName);
             listener.getLogger().format("Looking up %s%n", HyperlinkNote.encodeTo(repo.getHtmlUrl().toString(), fullName));
-            try {
-                repositoryUrl = repo.getHtmlUrl();
-                collaboratorNames = new HashSet<>(repo.getCollaboratorNames());
-            } catch (FileNotFoundException e) {
-                // not permitted
-                listener.getLogger().println("Not permitted to query list of collaborators, assuming none");
-                collaboratorNames = Collections.emptySet();
+            if (!repo.isPrivate()) { // random people cannot create pull requests to private repositories
```

Comment:
@stephenc I'm very sorry for bothering you on this, but I did not get the point and ask for additional clarification if you please.

What exactly are we trying to fix here? Say the person with r/o access can contribute to the repository with a PR, modify the `Jenkinsfile` and **deploy the PR HEAD to production environment**? Well, that would clearly be a sabotage. If this is the case then marking the repository private means that you provided access to people that already have some level of trust from you. 

Or are you saying that the changes in the PR to the `Jenkinsfile` will affect deployments **after the PR is merged** to target branch? Then there's usually a code review to identify and accept/decline that particular change.

I do not want to annoy you, just want to know the reasoning the project pursues when identifying trusted PRs.

Also reading the code further I've found that [trusted identification|https://github.com/jenkinsci/github-branch-source-plugin/blob/master/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java#L572] is used to [show the message in the log|https://github.com/jenkinsci/github-branch-source-plugin/blob/master/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java#L573] in that particular method all the way till [Observer.observe](https://github.com/jenkinsci/github-branch-source-plugin/blob/master/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java#L658) which does the heavy lifting. So that might be still be an overuse of precious api call to identify the collaborators just to show if the pull request is from trusted resource or not in the log of the organization scan.